src/EFCore.PG/Design/Internal/NpgsqlCSharpRuntimeAnnotationCodeGenerator.cs

```diff
@@ -28,6 +29,106 @@ public class NpgsqlCSharpRuntimeAnnotationCodeGenerator : RelationalCSharpRuntim
     {
     }
 
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public override bool Create(
+        CoreTypeMapping typeMapping,
+        CSharpRuntimeAnnotationCodeGeneratorParameters parameters,
+        ValueComparer? valueComparer = null,
+        ValueComparer? keyValueComparer = null,
+        ValueComparer? providerValueComparer = null)
+    {
+        var result = base.Create(typeMapping, parameters, valueComparer, keyValueComparer, providerValueComparer);
+
+        var mainBuilder = parameters.MainBuilder;
+
+        var npgsqlDbTypeBasedDefaultInstance = typeMapping switch
+        {
+            NpgsqlStringTypeMapping => NpgsqlStringTypeMapping.Default,
+            NpgsqlULongTypeMapping => NpgsqlULongTypeMapping.Default,
+            // NpgsqlMultirangeTypeMapping => NpgsqlMultirangeTypeMapping.Default,
+            _ => (INpgsqlTypeMapping?)null
+        };
+
+        if (npgsqlDbTypeBasedDefaultInstance is not null)
+        {
+            var npgsqlDbType = ((INpgsqlTypeMapping)typeMapping).NpgsqlDbType;
+
+            if (npgsqlDbType != npgsqlDbTypeBasedDefaultInstance.NpgsqlDbType)
+            {
+                mainBuilder.AppendLine(";");
+
+                mainBuilder.Append(
+                    $"{parameters.TargetName}.TypeMapping = (({typeMapping.GetType().Name}){parameters.TargetName}.TypeMapping).Clone(");
+
+                mainBuilder
```

Comment:
I prefer to always output the parameter name to make sure that existing code doesn't compile if there's a breaking change.

Reply:
Will do.